run/events-pubsub/Startup.cs

```diff
@@ -46,38 +45,47 @@ public void Configure(IApplicationBuilder app, IWebHostEnvironment env, ILogger<
         {
             endpoints.MapPost("/", async context =>
             {
-                logger.LogInformation("Handling HTTP POST");
-
-                using (var reader = new StreamReader(context.Request.Body))
+                var cloudEvent = await context.Request.ReadCloudEventAsync();
+                if (cloudEvent == null)
                 {
-                    var body = await reader.ReadToEndAsync();
-                    logger.LogInformation($"HTTP POST body: {body}");
+                    context.Response.StatusCode = 400;
+                    await context.Response.WriteAsync("Bad request: No CloudEvent received");
+                    return;
+                }
+                logger.LogInformation("Received CloudEvent\n" + GetEventLog(cloudEvent));
 
-                    dynamic cloudEventData = JsonConvert.DeserializeObject(body);
-                    if (cloudEventData == null)
-                    {
-                        context.Response.StatusCode = 400;
-                        await context.Response.WriteAsync("Bad request: No Pub/Sub message received");
-                        return;
-                    }
+                var messagePublishedData = CloudEventConverters.ConvertCloudEventData<MessagePublishedData>(cloudEvent);

```

Comment:
This one will most probably throw instead of just returing null if it can't convert. I'm OK with not using try-catch here, but then I think you souldn't check for null either, else, it gives the impression that if it can't convert, it will return null.

Reply:
I tried to reuse the existing code but you're right. Let me see what it actually does (throw or return null) and change the code accordingly.

Reply:
I checked and `CloudEventConverters.ConvertCloudEventData` does not throw or return null when the format is not right. It simply returns an empty `MessagePublishedData`. I changed the code to account for that.

Reply:
It's unfortunate that that's pretty much a requirement - events have to be able to add new fields, so we can't throw if we see fields we don't expect... which would be the case if something was "entirely the wrong format".
I believe it *will* throw if it's "somewhat the wrong format" - e.g. an expected-to-be-numeric field that had an object value.
Will think about whether there's anything else we can do about that...